mlrose_hiive/opt_probs/queens_opt.py

```diff
@@ -36,4 +36,6 @@ def __init__(self, length=None, fitness_fn=None, maximize=False,
         self.set_state(state)
 
     def can_stop(self):
+        if self.maximize == 1.0:
```

Comment:
```if self.maximize:```
is typically preferable to "if self.maximize == 1.0" for clarity and brevity's sake. The argument's default value is a boolean, and realistically we only care about its truth value.

Reply:
"The argument's default value is a boolean" - this is not true
look at line 47 in _OptProb initializer. 
maximize is passed in as a boolean, but self.maximize is not a boolean, I copied the code snippet below

Reply:
Is this right that the problem *has* to be stated as a maximization problem?

If so, then it's changed from when I took the course. If not, then I'm loathe to make a code change that isn't actually necessary.

Reply:
Yes, for the assignment this year all problems have to be stated as maximisation problems.